benchmarks/utilities/log_parser.hpp

```diff
@@ -80,6 +83,35 @@ inline std::ostream& operator<<(std::ostream& os, event const& e)
 
 inline uintptr_t hex_string_to_int(std::string const& s) { return std::stoll(s, nullptr, 16); }
 
+/**
+ * @brief Parse a log timestamp into a std::chrono::time_point
+ *
+ * @note currently unused. Seemed necessary for ordering but it appears the log currently
+ * is in timestamp order even for multithreaded logs.
+ * @note This function can be simplified with C++20 and later.
+ *
+ * @param str_time The input time in format "HH:MM:SS:us" where us is a 6 digits microseconds part
+ * of the current second. (This is the format rmm::mr::logging_resource_adaptor outputs)
+ * @return std::chrono::time_point<std::chrono::system_clock> Converted time point.
+ */
+inline std::chrono::time_point<std::chrono::system_clock> parse_time(std::string const& str_time)
+{
+  std::size_t current, previous = 0;
```

Comment:
`previous` is never updated, is this intentional?

Reply:
Fixed.